### PR TITLE
EC2_INSTANCE_NO_PUBLIC_IP

### PR DESCRIPTION
**Pull request by shikharj05**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

File: python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py [hunk 1]
```diff
@@ -9,6 +9,29 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
 # the specific language governing permissions and limitations under the License.
 
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  EC2_INSTANCE_NO_PUBLIC_IPv4_IP
```
Comment (jongogogo) [inline python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py]: please update the name of the rule

File: python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py [hunk 2]
```diff
@@ -9,6 +9,29 @@
...
 # the specific language governing permissions and limitations under the License.
 
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  EC2_INSTANCE_NO_PUBLIC_IPv4_IP
+Description:
+  Checks whether Amazon EC2 instances have a public IP association or not. The rule is NON_COMPLIANT if the publicIp field is present in the Amazon EC2 instance configuration item.
```
Comment (jongogogo) [inline python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py]: add the limitation on IPv4 here. Similar to INCOMING_SSH rule

**Comment by jongogogo** [comment]
Comment test

**Comment by jongogogo** [comment]
Can you please repush a change. The bot is working now. Thanks

File: python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+DEFAULT_RESOURCE_TYPE = 'AWS::EC2::Instance'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    if not re.findall('\\bpublicIp\\b', str(configuration_item)):
```
Comment (jongogogo) [inline python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py]: What if a tag is call `publicIp` or something else. This is need to be more elegant.

**Comment by shikharj05** [comment]
@jongogogo Thank you for your feedback, made the changes.(https://github.com/awslabs/aws-config-rules/pull/173/commits/b69e70141019ca3a6dd95bf463e21b4c35daa963) 
Now, the regex match is checked against 'networkInterfaces' (from the CI) only.

**Pull request merged**